themes/10up-theme/tailwind.config.js

```diff
@@ -1,4 +1,28 @@
-const tailwindProse = require('@tailwindcss/typography');
+const typographyPlugin = require('@tailwindcss/typography');
+const colors = require('tailwindcss/colors');
+const themeJSON = require('./theme.json');
+
+function getThemePalette() {
+	const { palette } = themeJSON.settings.color;
+
+	return Array.isArray(palette) ? palette : [];
+}
+
+function getThemeColors() {
```

Comment:
This is really cool! :) 

One thing to note though is that this will not get you all colors that are used in a `theme.json` file. 

Every block can override the global color palette and therefore also add its own color palette... 

```jsonc
{
   "settings": {
        "color": {
            palette: [ ... ] // ← these are the global colors
        }
    },
    "blocks": {
        "core/button": {
             "color": {
                 "palette": [ ... ] // ← a color that does not exist on the global level can be introduced here
             }
        }
    }
}
```


And it doesn't account for the fact that `theme.json` in its nature is not singular but instead has different layers that have more / less specificity. 

The default setup looks like this: 

![](https://gutenberg.10up.com/assets/images/global-styles-input-output-f1367794eb935ca00e653bcfb90f8a4a.png)


So the core / user-defined colors will not get picked up by the tailwind config. 

I'm not sure it really matters but just wanted to call it out.

Reply:
@fabiankaegy That's a great point and funny enough I was literally just reading about this today, great minds think alike!

After giving it some thought it seems unlikely that we'd have a custom block color that isn't an existing color elsewhere. For instance, if we have a brand guide from a client its unlikely they have a single color that is only used on a single block or something. 

I feel like it wouldn't be a huge hassle to include those values in this setup. I'm pretty sure the config is smart enough to strip any duplicates (ex: there's a global color that is also defined in a block's colors) which further reduces the life there. 

As you said I'm also not sure it really matters, but just playing devil's advocate for a second...can you think of any scenarios where we'd define a color like that for a block that isn't in the theme color palette that we want to use in our code?

Reply:
@iansvo The example I was thinking about is a Call to action block for example that might want to expose 4 color variants: 

- `alert-red`
- `warning-orange`
- `success-green`
- `info-blue`

I would not expect these colors to be available in the global palette because you may not want to allow for text to be colored in red throughout the rest of the page. 

Again I don't think this is mandatory and such an edge case that we can worry about it later. But I think there is a relatively straightforward way to refactor this code to iterate over all `blocks['block-name'].settings.color.palette` values, dedupe them and then use that for the color config :)